include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -959,135 +966,431 @@ void reduceSumScalingVector(common::Pipeline<comm::Communicator>& col_task_chain
+  };
 
-  auto sender =
-      ex::when_all(std::forward<KSender>(k), std::forward<RhoSender>(rho),
-                   ex::when_all_vector(tc.read(d)), ex::when_all_vector(tc.read(z)),
+  const std::size_t nthreads = [size = sz_loc_tiles.cols()]() {
+    const std::size_t min_workers = 1;
+    const std::size_t available_workers = getTridiagRank1NWorkers();
+    const std::size_t ideal_workers = util::ceilDiv(to_sizet(size), to_sizet(2));
+    return std::clamp(ideal_workers, min_workers, available_workers);
+  }();
+
+  ex::start_detached(
+      ex::when_all(ex::just(std::make_shared<pika::barrier<>>(nthreads)),
+                   std::forward<CommSender>(col_comm), std::forward<CommSender>(row_comm),
+                   std::forward<KSender>(k), std::forward<RhoSender>(rho),
+                   ex::when_all_vector(tc.read(d)), ex::when_all_vector(tc.readwrite(z)),
                    ex::when_all_vector(tc.readwrite(evals)), ex::when_all_vector(tc.readwrite(delta)),
-                   ex::when_all_vector(tc.readwrite(i2)), ex::when_all_vector(tc.readwrite(evecs)));
+                   ex::when_all_vector(tc.read(i2)), ex::when_all_vector(tc.readwrite(evecs)),
+                   // additional workspaces
+                   ex::just(std::vector<memory::MemoryView<T, Device::CPU>>()),
+                   ex::just(memory::MemoryView<T, Device::CPU>())) |
+      ex::transfer(di::getBackendScheduler<Backend::MC>(pika::execution::thread_priority::high)) |
+      ex::bulk(nthreads, [nthreads, n, n_subm_el_lc, m_subm_el_lc, i_begin, i_end, ij_begin_lc,
+                          sz_loc_tiles, dist, bcast_evals, allReduceInPlace_o](
+                             const std::size_t thread_idx, auto& barrier_ptr, auto& col_comm_wrapper,
+                             auto& row_comm_wrapper, const auto& k, const auto& rho,
+                             const auto& d_tiles_futs, auto& z_tiles, const auto& eval_tiles,
+                             const auto& delta_tile_arr, const auto& i2_tile_arr, const auto& evec_tiles,
+                             auto& ws_vecs, auto& ws_rowvec) {
+        using dlaf::comm::internal::transformMPI;
+
+        const dlaf::comm::Communicator& row_comm = row_comm_wrapper.get();
+        const dlaf::comm::Communicator& col_comm = col_comm_wrapper.get();
+
+        const std::size_t batch_size =
+            std::max<std::size_t>(2, util::ceilDiv(to_sizet(sz_loc_tiles.cols()), nthreads));
+        const SizeType begin = to_SizeType(thread_idx * batch_size);
+        const SizeType end = std::min(to_SizeType((thread_idx + 1) * batch_size), sz_loc_tiles.cols());
+
+        // STEP 0a: Fill ones for deflated Eigenvectors. (single-thread)
+        // Note: this step is completely independent from the rest, but it is small and it is going
+        // to be dropped soon.
+        // Note: use last threads that in principle should have less work to do
+        // TODO if chunked this can be run by multiple workers
```

Comment:
I would say that it will probably never get any improvement this block, because we are going to drop this as soon as we can improve the gemm. So I will remove this TODO 😉